### PR TITLE
Jim/74978/accordion component ts migration

### PR DESCRIPTION
**Pull request by jim-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [X] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by maryia-deriv — APPROVED** [review]
👍🏻

File: packages/components/src/components/accordion/accordion.tsx [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-const Accordion = ({ className, icon_close, icon_open, list }) => {
-    const [open_idx, setOpenIdx] = React.useState(null);
+type TAccordionProps = {
+    className: string;
+    icon_close?: string;
+    icon_open?: string;
+    list: Array<{
+        header: string;
+        content: React.ReactNode;
+    }>;
```
Comment (farzin-deriv) [inline packages/components/src/components/accordion/accordion.tsx]: @jim-deriv Just a suggestion, Maybe we can extract this to a separate type like `TAccordionItem` and also export it from this component.

**Review by farzin-deriv — APPROVED** [review]

File: packages/components/src/components/accordion/accordion.tsx [hunk 2]
```diff
@@ -1,11 +1,20 @@
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
 import { usePrevious } from '../../hooks';
 import Icon from '../icon';
 
-const Accordion = ({ className, icon_close, icon_open, list }) => {
-    const [open_idx, setOpenIdx] = React.useState(null);
+type TAccordionProps = {
+    className: string;
```
Comment (shayan-deriv) [inline packages/components/src/components/accordion/accordion.tsx]:
```suggestion
    className?: string;
```

**Review by shayan-deriv — APPROVED** [review]

File: packages/components/src/components/accordion/accordion.tsx (hunk 1 above)
Comment (jim-deriv) [reply packages/components/src/components/accordion/accordion.tsx]: Sure @farzin-fs

File: packages/components/src/components/accordion/accordion.tsx (hunk 2 above)
Comment (jim-deriv) [reply packages/components/src/components/accordion/accordion.tsx]: Sure @iman-fs

**Pull request merged**